write-fonts/src/tables/gvar.rs

```diff
@@ -934,4 +969,120 @@ mod tests {
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(-24, 0),
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::optional(0, 0),
+                ],
+                None,
+            ),
+            GlyphDeltas::new(
+                Tuple::new(vec![F2Dot14::from_f32(1.0), F2Dot14::from_f32(1.0)]),
+                vec![
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(26, 15),
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(46, 0),
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(1, 0),
+                ],
+                None,
+            ),
+        ]
+    }
+
+    // sanity checking my input data for two subsequent tests
+    #[test]
+    fn who_tests_the_testers() {
+        let variations = GlyphVariations::new(GlyphId::NOTDEF, make_39_bytes_of_variation_data());
+        // to get our real size we need to mock up the shared tuples:
+        let mut tupl_map = HashMap::new();
+        tupl_map.insert(&variations.variations[0].peak_tuple, 1);
+        tupl_map.insert(&variations.variations[1].peak_tuple, 2);
+
+        let built = variations.clone().build(&tupl_map);
+        // we need an odd number for this to make sense
+        assert_eq!(built.length, 31);
+    }
+
+    #[test]
+    fn prefer_short_offsets() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        const N_GLYPHS: u16 = 5;
+        let test_data = make_39_bytes_of_variation_data();
+        let a_small_number_of_variations = (0..N_GLYPHS)
+            .map(|i| GlyphVariations::new(GlyphId::new(i), test_data.clone()))
+            .collect();
+
+        let gvar = Gvar::new(a_small_number_of_variations).unwrap();
+        assert_eq!(gvar.compute_flags(), GvarFlags::empty());
+
+        let writer = gvar.compile_variation_data();
+        let mut sink = TableWriter::default();
+        writer.write_into(&mut sink);
+
+        let bytes = sink.into_data().bytes;
+        let expected_len = (N_GLYPHS + 1) as usize * u16::RAW_BYTE_LEN // offsets
+                             + 32 * N_GLYPHS as usize; // rounded size of each glyph, * 5 glyphs
+        assert_eq!(bytes.len(), expected_len);
+
+        let dumped = crate::dump_table(&gvar).unwrap();
+        let loaded = read_fonts::tables::gvar::Gvar::read(FontData::new(&dumped)).unwrap();
+
+        assert_eq!(loaded.glyph_count(), N_GLYPHS);
+        assert_eq!(loaded.flags(), GvarFlags::empty());
+        assert_eq!(
+            loaded
+                .glyph_variation_data_offsets()
+                .iter()
+                .map(|off| off.unwrap().get())
+                .collect::<Vec<_>>(),
+            vec![0u32, 32, 64, 96, 128, 160]
+        );
+    }
+
+    #[test]
+    fn use_long_offsets_when_necessary() {
+        // 2**16 * 2 / (31 + 1 padding) (bytes per tuple) = 4096 so this just overflows
```

Comment:
I suggest you test the most possible short offsets as well, for example loop over N_GLYPHS 4099, 4100 and confirm that their respective flags are SHORT_OFFSETS, LONG_OFFSETS with the expected sizes